### ISSUE TITLE
Stabilizing serialization for ranges, domains, and iterators in non-default formats

### ISSUE THREAD

**Issue by benharsh** [chapel-lang/chapel#23394]
PR #23392 implemented some unstable options for serializing and deserializing ranges, domains, and iterators when using a non-default serializer/deserializer. The choices made at the time were:
- ranges print as strings containing the default-formatting text. For example, in JSON you would see ``"1..10"``
- rectangular domains print as lists of their dimensions, e.g. ``["1..10", "1..10"]``
- iterator records print as lists

Are these choices ones we think are suitable to stabilize? Or are there other representations we should consider?

**Comment by benharsh** [comment]
A couple of things to consider:
- domains may be more suitable as tuples
- iterator records may be more suitable as arrays, at least in the case of iterators with multidimensional "shape"
- while printing ranges as strings seems like a useful option in most textual formats, it's a bit wasteful in binary formats

**Comment by bradcray** [comment]
I'm way out of date with the serializer work, but my intuition for serializing a range in a format that was non-textual or not intended to be intuitive would be to write out their (low, high, stride, alignment) 4-tuple.

**Comment by mppf** [comment]
> while printing ranges as strings seems like a useful option in most textual formats, it's a bit wasteful in binary formats

Yeah I'd go with something like the 4-tuple Brad mentioned for binary formats, and something like the string / like what we write in a range literal for textual formats.

**Comment by mppf** [comment]
The current approach leads to some problems with serializing a domain (or range) with chplSerializer but I've made #24123 about that issue specifically.
